### PR TITLE
RUN: Show stdout for failed tests

### PR DESCRIPTION
**Pull request by mchernyavsky**
Closes https://github.com/intellij-rust/intellij-rust/issues/3030.

### CODE REVIEW

File: src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestEventsConverter.kt [hunk 1]
```diff
@@ -257,10 +258,22 @@ class CargoTestEventsConverter(
...
+                .addAttribute("out", stdout)
+
         private fun createTestCountMessage(testCount: String): ServiceMessageBuilder =
             ServiceMessageBuilder("testCount")
                 .addAttribute("count", testCount)
 
+        private fun parseFailedTestOutput(output: String): FailedTestOutput {
+            val partitionPredicate: (String) -> Boolean = { !it.trimStart().startsWith("thread", true) }
+            val stdout = output.lineSequence().takeWhile(partitionPredicate).joinToString("\n")
+            val failedMessage = output.lineSequence().dropWhile(partitionPredicate).joinToString("\n")
```
Comment (Undin) [inline src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestEventsConverter.kt]: Looks like you can use `Sequence.partition` here

Comment (mchernyavsky) [reply src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestEventsConverter.kt]: This name is deceptive :)
> Splits the original sequence into pair of lists, where first list contains elements for which predicate yielded `true`, while second list contains elements for which predicate yielded `false`.

**Comment by matklad** [comment]
Hm, I think I disagree about this change in behavior :-) Original divergence from Cargo's defaults was very intentional. 

Cargo's interface is tailored for a command line usage, and, on the command line, you typically execute tests as `cargo test`. This runs all of the tests, without any UI affordances for a fancier output that a "wall of text". In this situation, hiding stdout/stderr by default can be reasonable (although I personally still find it more confusing than helpful). 

In IDE, the situation is quite different. *First*, the user typically runs a subset of the tests, or a single test only. Because of this, the total amount of information is much smaller. *Second*, we can smartly group tests by module, crate, etc, which makes it easy to slice the "wall of text" into manageable pieces. *Third*, test output view is a rather non-central panel in IDE (unlike a typical full-screen terminal), so flooding it with lots of messages it not a problem in the first place.


OTOH, *not* showing the output is quite confusing, especially for beginners. So, I think that it makes sense to stick with `RUST_LOG=short` `--no-capture` *by default*, unless many users explicitly demand the silent behavior :)

**Comment by matklad** [comment]
Or maybe I am misunderstanding what this PR does? To reforumulate, if I run

```
#[test]
fn test_stdout() {
    println!("hello, World")
}
```

in IntelliJ Rust today, I don't see "Hello, World" anywhere, and it seems wrong to me

**Comment by mchernyavsky** [comment]
@matklad Without `--nocapture` flag, cargo shows stdout/stderr for failed tests only. We don't use `--nocapture` because of https://github.com/rust-lang/rust/issues/54669. It will be fixed in https://github.com/intellij-rust/intellij-rust/pull/4139.

**Pull request merged**